### DISCUSSION TITLE
ckb 0.105.0 (1d3af67 2022-10-25)

### DISCUSSION THREAD

**Discussion by doitian** [nervosnetwork/ckb#3655] [Announcements]
### Highlights

**Attention**: This version requires database migration, which may take half an hour or more.

This version will activate CKB Light Client support in testnet via soft fork. See more details in #3652 and the following RFCs:

- nervosnetwork/rfcs#349
- nervosnetwork/rfcs#370
- nervosnetwork/rfcs#375

### Compatability Table

| Network | Consensus Version        | Minimal CKB Version |
|---------|------------------------- | ------------------- |
| dev     | ckb2021 since epoch 0    |                     |
| testnet | ckb2021 since epoch 3113 | v0.101.0            |
| mainnet | ckb2021 since epoch 5414 | v0.103.0            |

See [how to connect to various chains](https://github.com/nervosnetwork/ckb/wiki/Chains) in Wiki.

### Downloads

| OS                | Arch  | Runtime Dependencies | Package                 | Sign                |
| ----------------- | ----- | -------------------- | ----------------------- | ------------------- |
| macOS             | x64   | macOS 10.15 or above | [zip][macos-zip]        | [PGP][macos-pgp]    |
| macOS  (Portable) | x64   | macOS 10.15 or above | [zip][macos-p-zip]      | [PGP][macos-p-pgp]  |
| Linux             | x64   | glibc, libstdc++     | [tarball][linux-zip]    | [PGP][linux-pgp]    |
| Linux  (Portable) | x64   | glibc, libstdc++     | [tarball][linux-p-zip]  | [PGP][linux-p-pgp]  |
| CentOS            | x64   | glibc, libstdc++     | [tarball][centos-zip]   | [PGP][centos-pgp]   |
| CentOS (Portable) | x64   | glibc, libstdc++     | [tarball][centos-p-zip] | [PGP][centos-p-pgp] |
| Windows           | x64   | VC++ Redistributable | [zip][windows-zip]      | [PGP][windows-pgp]  |
| Linux             | ARM64 | glibc, libstdc++     | [tarball][arm-zip]      | [PGP][arm-pgp]      |
| Docker            |       |                      | [dockerhub][]           |                     |

The `.asc` files are signatures. It is wise and more secure to check out for the files [integrity](https://github.com/nervosnetwork/ckb/blob/develop/docs/integrity-check.md).

Try the portable variants if the binaries complains about illegal instructions.

[macos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-apple-darwin.zip
[macos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-apple-darwin.zip.asc
[macos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-apple-darwin-portable.zip
[macos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-apple-darwin-portable.zip.asc
[linux-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-linux-gnu.tar.gz
[linux-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-linux-gnu.tar.gz.asc
[linux-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-linux-gnu-portable.tar.gz
[linux-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-linux-gnu-portable.tar.gz.asc
[centos-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-centos-gnu.tar.gz
[centos-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-centos-gnu.tar.gz.asc
[centos-p-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-centos-gnu-portable.tar.gz
[centos-p-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-unknown-centos-gnu-portable.tar.gz.asc
[windows-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-pc-windows-msvc.zip
[windows-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_x86_64-pc-windows-msvc.zip.asc
[arm-zip]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_aarch64-unknown-linux-gnu.tar.gz
[arm-pgp]: https://github.com/nervosnetwork/ckb/releases/download/v0.105.0/ckb_v0.105.0_aarch64-unknown-linux-gnu.tar.gz.asc
[dockerhub]: https://hub.docker.com/r/nervos/ckb/tags

<details><summary>Changes since v0.104.0</summary>

### Features

* #3544: Add GetTransactions message to light client protocol (@quake)
* #3595: P2P Flags features (@driftluo)
* #3565: ChainRpcImpl: support return raw molecule hex bytes for `get_transaction` rpc method (@eval-exec)
* #3564: Adding JSON output support for `ckb list-hashes` (@code-monad)
* #3609: Merge indexer (@zhangsoledad)
* #3627: Bump ckb-vm to v0.21.7 (@mohanson)
* #3631 **light-client:** Add light client support to ckb full node (@quake)
* #3515 **light-client:** Light client softfork (@zhangsoledad)
* #3643 **light-client:** Activation parameters for testnet lightclient (@zhangsoledad)

### Bug Fixes

* #3516: Fix atomic fetch update ordering (@driftluo)
* #3566: Fix tcp reuse bind (@driftluo)
* #3642: Network flags should be configured accordingly (@quake)

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/nervosnetwork/ckb/releases/tag/v0.105.0'>ckb 0.105.0 (1d3af67 2022-10-25)</a>.</em>

**Comment by joshyates1980** [comment]
Database migration? Does this mean I will need to delete the "data" folder for the update? Normally I will delete only the binary files when I apply the ckb update.

**Reply by driftluo** [reply]
no, just use the new binary run `ckb migration -C your_data_path` and wait for it finishes, then run `ckb run -C your_data_path`

It is recommended that you wait for 0.105.1 to be released before doing this, 0.105.0 consumes a lot of memory on migration, about 5-7G, and takes a long time, this will be improved in 0.105.1
